crates/starknet-os/src/hints/compiled_class.rs

```diff
@@ -166,4 +192,162 @@ mod tests {
         let res = assert_end_of_bytecode_segments(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants);
         assert!(res.is_ok());
     }
+
+    #[test]
+    fn test_set_ap_to_segment_hash() {
+        use num_bigint::BigUint;
+
+        let mut vm = VirtualMachine::new(false);
+        vm.add_memory_segment();
+        vm.add_memory_segment();
+        vm.set_fp(2);
+
+        let ap_tracking = ApTracking::new();
+        let constants = HashMap::new();
+        let ids_data = HashMap::new();
+
+        let mut exec_scopes: ExecutionScopes = Default::default();
+
+        // Execution scopes must have a BytecodeSegmentStructureImpl inserted. We insert one that has one
+        let segments = vec![BytecodeSegment {
+            segment_length: Length(1),
+            is_used: false,
+            inner_structure: BytecodeSegmentStructureImpl::Leaf(BytecodeLeaf { data: vec![BigUint::from(1u8)] }),
+        }];
+        let segment_structure = BytecodeSegmentStructureImpl::SegmentedNode(BytecodeSegmentedNode { segments });
+        exec_scopes.insert_box(vars::scopes::BYTECODE_SEGMENT_STRUCTURE, any_box!(segment_structure));
+
+        set_ap_to_segment_hash(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants).unwrap();
+
+        // Read hash and compare with expected hash
+        let hash = vm.get_integer(vm.get_ap()).unwrap().into_owned();
+        assert_eq!(hex::encode(hash.to_bytes_be()), "064b3967128647f5db91e107897e7e6d72f2a06f35d01d19055a7f85c85e65ba");
+    }
+
+    #[test]
+    fn test_hash_bytecode_leaf() {
+        // Reference hashes were taken from cairo-lang (compiled_class_hash_test.py)
```

Comment:
These don't look like the class hashes from [compiled_class_hash_test.py](https://github.com/starkware-libs/cairo-lang/blob/ab6d079f0265f903630aac9287e93d9e7cfb6fcf/src/starkware/starknet/core/os/contract_class/compiled_class_hash_test.py#L25).

Reply:
They are different because the code is hashing`BytecodeSegment` structs, not compiled classes.

I added these lines to the Python tests to extract the information I needed. However, in the hint test, I used a simpler struct that was also extracted from the Python code.
```
 assert bytecode_segment_structure.bytecode_with_skipped_segments() == expected_bytecode
 print("Bytecode struct: ", bytecode_segment_structure)
 print("Bytecode struct hash: ", hex(bytecode_segment_structure.hash()))
```

Reply:
Okay, could you reword this comment to make this clearer?